test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -802,6 +802,500 @@ test.describe( 'Navigation block', () => {
 	} );
 } );
 
+test.describe( 'Navigation block - Frontend interactivity', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await requestUtils.deleteAllPages();
+		await requestUtils.deleteAllMenus();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await requestUtils.deleteAllPages();
+		await requestUtils.deleteAllMenus();
+	} );
+
+	test.describe( 'Overlay menu', () => {
+		test.beforeEach( async ( { admin, editor, requestUtils } ) => {
+			await admin.visitSiteEditor( {
+				postId: 'emptytheme//header',
+				postType: 'wp_template_part',
+			} );
+			await editor.canvas.click( 'body' );
+			await requestUtils.createNavigationMenu( {
+				title: 'Hidden menu',
+				content: `
+					<!-- wp:navigation-link {"label":"Item 1","type":"custom","url":"http://www.wordpress.org/","isTopLevelLink":true} /-->
+					<!-- wp:navigation-link {"label":"Item 2","type":"custom","url":"http://www.wordpress.org/","isTopLevelLink":true} /-->
+					`,
+			} );
+			await editor.insertBlock( {
+				name: 'core/navigation',
+				attributes: { overlayMenu: 'always' },
+			} );
+			await editor.saveSiteEditorEntities();
+		} );
+
+		test( 'overlay menu opens on click on open menu button', async ( {
+			page,
+		} ) => {
+			await page.goto( '/' );
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
```

Comment:
Any reason why we're not using role selectors here (and throughout the file below)? If it's because the element isn't accessible yet, we should probably add comments above so that people don't blindly copy-paste them 😅 .

Reply:
Where you can find out more details about using role selectors?

Reply:
I was just changing them, sorry. I just pushed a commit changing all the `locator` for `getByRole`. Thanks for the advice 🙂

Reply:
> Where can you find out more details about using role selectors?

There's a section about using accessible selectors in the [best practices guideline](https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors/code/e2e#use-accessible-selectors). The official [doc](https://playwright.dev/docs/locators) is also great for finding out more details! :)

> I was just changing them, sorry.

No worries! Thanks! ❤️